drivers/backup/portworx/portworx.go

```diff
@@ -1092,6 +1097,310 @@ func (p *portworx) GetBackupUID(ctx context.Context, orgID, backupName string) (
+		Name:  ruleName,
+		Uid:   uid,
+	}
+	_, err1 := p.InspectRule(ctx, RuleInspectReq)
+	if err1 != nil {
+		log.Errorf("Failed to validate the created rule with Error: [%v]", err)
+		return false, uid, err1
+	}
+	return true, uid, nil
+}
+
+func (p *portworx) CreateIntervalSchedulePolicy(retain int64, min int64, incrCount uint64) *api.SchedulePolicyInfo {
+	SchedulePolicy := &api.SchedulePolicyInfo{
+		Interval: &api.SchedulePolicyInfo_IntervalPolicy{
+			Retain:  retain,
+			Minutes: min,
+			IncrementalCount: &api.SchedulePolicyInfo_IncrementalCount{
+				Count: incrCount,
+			},
+		},
+	}
+	return SchedulePolicy
+}
+
+func (p *portworx) CreateDailySchedulePolicy(retain int64, time string, incrCount uint64) *api.SchedulePolicyInfo {
+	SchedulePolicy := &api.SchedulePolicyInfo{
+		Daily: &api.SchedulePolicyInfo_DailyPolicy{
+			Retain: retain,
+			Time:   time,
+			IncrementalCount: &api.SchedulePolicyInfo_IncrementalCount{
+				Count: incrCount,
+			},
+		},
+	}
+	return SchedulePolicy
+}
+
+func (p *portworx) CreateWeeklySchedulePolicy(retain int64, day backup.Weekday, time string, incrCount uint64) *api.SchedulePolicyInfo {
+
+	SchedulePolicy := &api.SchedulePolicyInfo{
+		Weekly: &api.SchedulePolicyInfo_WeeklyPolicy{
+			Retain: retain,
+			Day:    string(day),
+			Time:   time,
+			IncrementalCount: &api.SchedulePolicyInfo_IncrementalCount{
+				Count: incrCount,
+			},
+		},
+	}
+	return SchedulePolicy
+}
+
+func (p *portworx) CreateMonthlySchedulePolicy(retain int64, date int64, time string, incrCount uint64) *api.SchedulePolicyInfo {
+	SchedulePolicy := &api.SchedulePolicyInfo{
+		Monthly: &api.SchedulePolicyInfo_MonthlyPolicy{
+			Retain: retain,
+			Date:   date,
+			Time:   time,
+			IncrementalCount: &api.SchedulePolicyInfo_IncrementalCount{
+				Count: incrCount,
+			},
+		},
+	}
+	return SchedulePolicy
+}
+
+func (p *portworx) BackupSchedulePolicy(name string, uid string, orgId string, schedulePolicyInfo *api.SchedulePolicyInfo) error {
+	log.InfoD("Create Backup Schedule Policy")
+	ctx, err := backup.GetAdminCtxFromSecret()
+	if err != nil {
+		log.Errorf("Failed to fetch px-central-admin ctx: [%v]", err)
+		return err
+	}
+	schedulePolicyCreateRequest := &api.SchedulePolicyCreateRequest{
+		CreateMetadata: &api.CreateMetadata{
+			Name:  name,
+			Uid:   uid,
+			OrgId: orgId,
+		},
+		SchedulePolicy: schedulePolicyInfo,
+	}
+	_, err = p.CreateSchedulePolicy(ctx, schedulePolicyCreateRequest)
+	if err != nil {
+		log.Errorf("Error in creating schedule policy is +%v", err)
+		return err
+	}
+	return nil
+}
+
+func (p *portworx) RegisterBackupCluster(orgID, clusterName, uid string) (api.ClusterInfo_StatusInfo_Status, string) {
+	ctx, err := backup.GetAdminCtxFromSecret()
+	log.FailOnError(err, "Failed to fetch px-central-admin ctx")
+	clusterReq := &api.ClusterInspectRequest{OrgId: orgID, Name: clusterName, IncludeSecrets: true}
+	clusterResp, err := p.InspectCluster(ctx, clusterReq)
+	log.FailOnError(err, "Cluster Object nil")
```

Comment:
you can request params here, so that it will be helpfil in case of error what are the params passed

Reply:
done!